pkg/storaged/resize.jsx

```diff
@@ -0,0 +1,450 @@
+            return Promise.resolve();
+        }
+    }
+
+    function crypto_resize() {
+        if (crypto) {
+            const opts = { };
+            if (passphrase)
+                opts.passphrase = { t: "s", v: passphrase };
+            return crypto.Resize(size - crypto_overhead, opts);
+        } else {
+            return Promise.resolve();
+        }
+    }
+
+    function lvol_or_part_resize() {
+        if (size != orig_size) {
+            if (lvol_or_part.iface == "org.freedesktop.UDisks2.LogicalVolume")
+                return lvol_or_part.Resize(size, { });
+            else
+                return lvol_or_part.Resize(size, { }); // It's the SAME!
+        } else
+            return Promise.resolve();
+    }
+
+    console.log("++", fsys);
+
+    if (fsys && !fsys.Resize) {
+        // Fallback for old versions of UDisks.  This doesn't handle encrypted volumes.
+        if (size != orig_size && lvol_or_part.iface == "org.freedesktop.UDisks2.LogicalVolume") {
+            return lvol_or_part.Resize(size, { resize_fsys: { t: 'b', v: true } });
+        }
+    } else {
+        if (size < orig_size) {
+            return fsys_resize().then(crypto_resize)
+                    .then(lvol_or_part_resize);
+        } else if (size >= orig_size) {
+            return lvol_or_part_resize().then(crypto_resize)
+                    .then(fsys_resize);
+        }
+    }
+}
+
+export function get_resize_info(client, block, to_fit) {
+    let info, shrink_excuse, grow_excuse;
+
+    if (block) {
+        if (block.IdUsage == 'crypto' && client.blocks_crypto[block.path]) {
+            const encrypted = client.blocks_crypto[block.path];
+            const cleartext = client.blocks_cleartext[block.path];
+
+            if (!encrypted.Resize) {
+                info = { };
+                shrink_excuse = grow_excuse = _("Encrypted volumes can not be resized here.");
+            } else if (!cleartext) {
+                info = { };
+                shrink_excuse = grow_excuse = _("Encrypted volumes need to be unlocked before they can be resized.");
+            } else {
+                return get_resize_info(client, cleartext, to_fit);
+            }
+        } else if (block.IdUsage == 'filesystem') {
+            info = client.fsys_info && client.fsys_info[block.IdType];
+
+            if (!info) {
+                info = { };
+                shrink_excuse = grow_excuse = cockpit.format(_("$0 filesystems can not be resized here."),
+                                                             block.IdType);
+            } else {
+                if (!info.can_shrink)
+                    shrink_excuse = cockpit.format(_("$0 filesystems can not be made smaller."),
+                                                   block.IdType);
+                if (!info.can_grow)
+                    grow_excuse = cockpit.format(_("$0 filesystems can not be made larger."),
+                                                 block.IdType);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19385-20230925-111150-44391ef2-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)